### PR TITLE
spans: add profile_id column

### PR DESCRIPTION
**Pull request by dbanda**
add profile_id column to spans.

### CODE REVIEW

File: snuba/snuba_migrations/spans/0007_spans_add_profile_id.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+                ),
+                target=OperationTarget.LOCAL,
+                after="span_id",
+            ),
+            operations.AddColumn(
+                storage_set=storage_set_name,
+                table_name=dist_table_name,
+                column=Column(
+                    "profile_id",
+                    UInt(64),
```
Comment (nikhars) [inline snuba/snuba_migrations/spans/0007_spans_add_profile_id.py:40]: On the transactions table, profile_id field is a UUID [field](https://github.com/getsentry/snuba/blob/master/snuba/snuba_migrations/transactions/0018_transactions_add_profile_id.py). This should probably be the same. Can you confirm @phacops ?

Comment (nikhars) [reply snuba/snuba_migrations/spans/0007_spans_add_profile_id.py:40]: As per conversation with Pierre offline, profile_id needs to be UUID field. Let's change it accordingly.

Comment (phacops) [reply snuba/snuba_migrations/spans/0007_spans_add_profile_id.py:40]: Indeed, it should be a `Uuid` column.

Comment (dbanda) [reply snuba/snuba_migrations/spans/0007_spans_add_profile_id.py]: ok fixed it.

File: snuba/snuba_migrations/spans/0007_spans_add_profile_id.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+    blocking = False
+
+    def forwards_ops(self) -> Sequence[SqlOperation]:
+        return [
+            operations.AddColumn(
+                storage_set=storage_set_name,
+                table_name=local_table_name,
+                column=Column(
+                    "profile_id", UUID(Modifiers(nullable=True, codecs=["ZSTD(1)"]))
```
Comment (nikhars) [inline snuba/snuba_migrations/spans/0007_spans_add_profile_id.py:27]: Not sure how much zstd compression would benefit on a UUID column. Not a blocker for the PR. Was just curious to know.

**Review by nikhars — APPROVED** [review]

File: snuba/snuba_migrations/spans/0007_spans_add_profile_id.py (hunk 2 above)
Comment (dbanda) [reply snuba/snuba_migrations/spans/0007_spans_add_profile_id.py:27]: Yeah there isn't much benefit. In the docs they mentioned using ZSTD over LZ4 by default is better.

**Pull request merged**